stdlib/list.ml

```diff
@@ -56,14 +56,18 @@ let rec rev_append l1 l2 =
 
 let rev l = rev_append l []
 
-let rec init_aux acc n f =
-  if n < 0 then acc else
-  init_aux ((f n)::acc) (n-1) f
+let rec init_tailrec_aux acc i n f =
+  if i >= n then acc
+  else init_tailrec_aux ((f i)::acc) (i+1) (n) f
+
+let rec init_aux i n f =
+  if i >= n then []
+  else (f i)::init_aux (i+1) n f
```

Comment:
You can test the behavior before and after the threshold (currently 10_000) using a function such as 

```ocaml

fun i -> if i < 10 || i > 10_000 && i < 10_010 then print_endline (string_of_int i)

```


It would allow to cover most branches without committing a giant test output file.